### PR TITLE
Add documentation for 7010 errors

### PR DESCRIPTION
**Pull request by Morriar**
image

### CODE REVIEW

File: website/docs/error-reference.md [hunk 1]
```diff
@@ -1126,6 +1126,38 @@ def publish_item(name, category = nil)
...
+
+  # ...
+end
+
+MyMap[String].new # error: Wrong number of type parameters for `MyMap`. Expected: `2`, got: `1`
+```
+
+Unless a type member was `fixed`, it is always required to pass the correct
+amount of type arguments. `T.untyped` can also be used is the type is not
+relevant at this point:
```
Comment (vaporyhumo) [inline website/docs/error-reference.md:1153]:
```suggestion
amount of type arguments. `T.untyped` can also be used as the type is not
relevant at this point:
```

Comment (Morriar) [reply website/docs/error-reference.md]: I meant `if`, thanks! 👍

File: website/docs/error-reference.md [hunk 2]
```diff
@@ -1126,6 +1126,38 @@ def publish_item(name, category = nil)
 end
 ```
 
+## 7010
+
+Sorbet found a reference to a generic type with the wrong number of type
+arrguments.
+
+Here we defined `MyMap` as a generic class expecting two type parameters
+`KeyType` and `ValueType` but we try to instanciate it with only one type
```
Comment (jez) [inline website/docs/error-reference.md:1135]:
```suggestion
`KeyType` and `ValueType` but we try to instantiate it with only one type
```

**Review by jez — APPROVED** [review]
You have conflicts with another PR (probably #4448)

**Comment by Morriar** [comment]
Rebased on master 👍

**Comment by Morriar** [comment]
@jez I think this one can be safely merged 👍

**Pull request merged**